src/app/shared/components/workshop-card/workshop-card.component.ts

```diff
@@ -58,6 +65,10 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
         this.favoriteWorkshopId = this.favoriteWorkshops?.find(item => item.workshopId === this.workshop?.workshopId);
       });
     this.isFavorite = !!this.favoriteWorkshopId;
+
+    this.provider$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((provider)=> this.isProvider = !!provider);
```

Comment:
you do not need it. you don't use the inctance of provider (you don't use its id/name/ etc). you just need the role for detecting the view type. so use please the useRole input